### ISSUE TITLE
Simplificação do TripleBuilder

### ISSUE THREAD

**Issue by pretodev** [Flutterando/triple_pattern#40]
Criei um shortcut que venho utilizando em meus projetos para diminuir a complexidade da arvore de widgets quando utilizo o Triple. Basicamente, é uma extension com um método que cria um TripleBuilder.  Segue o código:

```dart
import 'package:flutter/widgets.dart';
import 'package:flutter_triple/flutter_triple.dart';

typedef Widget TripleBuilderFunc<TError extends Object, TState extends Object>(
    Triple<TError, TState> snap);

typedef bool Filter<TError extends Object, TState extends Object>(
    Triple<TError, TState> triple);

typedef dynamic Distinct<TError extends Object, TState extends Object>(
    Triple<TError, TState> state);

extension StoreExtension<TError extends Object, TState extends Object>
    on StreamStore<TError, TState> {
  /// Create a TripleBuilder Widget from store
  Widget obx(
    TripleBuilderFunc<TError, TState> builder, {
    Filter? filter,
    Distinct? distinct,
  }) {
    return TripleBuilder(
      store: this,
      builder: (context, store) => builder(store as Triple<TError, TState>),
      filter: filter,
      distinct: distinct,
    );
  }
}
```

Sua utilização no código cliente fica assim:

```dart
IconButton(
  icon: store.obx(
    (snap) => Icon(
      snap.state.flashOn ? Icons.flash_on : Icons.flash_off,
    ),
  ),
  onPressed: store.toogleFlash,
),
```

Gostaria de saber se vocês acham interessante eu realizar um Pull request com este código para o projeto.

**Comment by jacobaraujo7** [comment]
Obrigado pela contribuição! Com certeza dará um bom debate.

Tenho duas visões sobre o assunto:
- A primeira seria adicionar o context no callback apenas por boas práticas, e também melhorar o nome do método, pois Obx acaba não significando muita coisa.

- A segunda seria adicionar essa extension em um package separado, preservando o padrão proposto pela documentação.

A minha sugestao(concatenando essas duas visoes) seria adicionar um método no objeto Store chamado "listen", que faria par com o observer,  porém retornaria todas as mudanças de estado e seus segmentos(como no TripleBuilder). O método listen poderia retornar um valor passado em generics:

```dart
T listen<T>(T Function(Triple triple);
```
Podendo ser usado:
```dart
store.listen<Widget>((value) => Text(value.state));
```

Caso essa sugestão seja apreciada, é uma alteração no Triple e não no Flutter_triple, e mesmo que não tenha breakchanges, haverá uma alteração no padrão.

**Comment by pretodev** [comment]
Concordo com você Jacob, isso seria uma solução para outro package para não ocorrer quebra no padrão. 

Quanto ao nome do método acredito poderia ser "map" já a ideia é fazer um mapeamento do estado para um novo objeto.

**Comment by jacobaraujo7** [comment]
Ou talvez when. Mas teria que ser sem reatividade.
Vou trabalhar nisso
